src/domain/challenge/challenge.service.ts

```diff
@@ -1,8 +1,13 @@
-import { Inject, Injectable, Logger } from '@nestjs/common';
+import { Inject, Injectable, LoggerService } from '@nestjs/common';
 import { InjectRepository } from '@nestjs/typeorm';
 import { WINSTON_MODULE_NEST_PROVIDER } from 'nest-winston';
-import { Repository } from 'typeorm';
-import { LogContexts } from '../../utils/logging/logging.contexts';
+import { Entity, Repository } from 'typeorm';
+import { EntityNotFoundException } from '../../utils/error-handling/exceptions/entity.not.found.exception';
+import { EntityNotInitializedException } from '../../utils/error-handling/exceptions/entity.not.initialized.exception';
+import { GroupNotInitializedException } from '../../utils/error-handling/exceptions/group.not.initialized.exception';
```

Comment:
Why both a Group + Entity not initialised exceptions? I would have thought 1 was enough?

Reply:
That's up for discussion. I feel group management is significant part of our community management logic and deserves its own category. EntityNotInitializedException is much more generic.